src/NzbDrone.Core/Datastore/Migration/074_add_add_options_to_series.cs

```diff
@@ -0,0 +1,14 @@
+﻿using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(74)]
+    public class add_add_options_to_series : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("Series").AddColumn("AddOptions").AsString().Nullable();
```

Comment:
I think thats a separate table/concept. I think we still need to be able to chain different events, like we're doing here.

The other case is when we do a series/episode refresh that adds a new already aired series (which I had to disable because of this case):

1. episode is missing from Sonarr
2. Gets added to TVDB, but Skyhook cache delays it getting into Sonarr
3. User grabs file manually and imports it
4. Sonarr picks up new episode
5. Event is triggered that contains newly added past episode
6. Sonarr searches for it because its missing
7. Sonarr queues episode in download client
8. Sonarr scans disk and find existing file
9. Queued download finishes and imports overwriting the existing file

Ultimately, the user downloaded the same file 2x for no reason, Sonarr should do the disk scan before the search, but that requires two events to be handled with some sort of storage in the middle.

1. Refresh event triggered
2. Sonarr stores event with newly added episodes in a "queue" (if no newly added episodes this step is skipped)
3. Disk scan complete event is triggered (New episode files from disk are now in DB)
4. Sonarr checks the "queue" if there is no stored event from the refresh, its done, otherwise start step 5
5. Sonarr searches for newly added + missing episode
6. Finds release
7. Queues release

The command queue PR still makes a lot of sense because it eliminates a few minor issues we have today and gives us better command tracking.